### ISSUE TITLE
Add labels to all form inputs for accessibility

### ISSUE THREAD

**Issue by jenweber** [ember-learn/guides-source#798]
For accessibility, all form inputs should have labels. The `for` on the label should match the `id` on the input. Add them to all places that inputs are used in the `guides/release` directory. This PR should branch from master and be merged back into master.

Before:

```
{{input value=something}}
```

After:

```
<label for="my-id">Something descriptive: </label>
{{input value=something id="my-id"}}
```

**Comment by rajasegar** [comment]
@jenweber Added labels to input fields in the following 5 places where we have used {{input}} fields
- Components / Defining a Component
- Components / The Component Life Cycle
- Components / Triggering changes with actions
- Templates / Input helpers
- Testing / Testing components
